packages/messaging/src/models/token-details-model.ts

```diff
@@ -119,190 +107,114 @@ export class TokenDetailsModel extends DBInterface {
-          request.onerror = event => {
-            reject((event.target as IDBRequest).error);
-          };
-          request.onsuccess = event => {
-            resolve();
-          };
-        });
-      });
+    const details = {
+      swScope: tokenDetails.swScope,
+      vapidKey: arrayBufferToBase64(tokenDetails.vapidKey as Uint8Array),
+      endpoint: tokenDetails.subscription.endpoint,
+      auth: arrayBufferToBase64(tokenDetails.subscription.getKey('auth')),
+      p256dh: arrayBufferToBase64(tokenDetails.subscription.getKey('p256dh')),
+      fcmSenderId: tokenDetails.fcmSenderId,
+      fcmToken: tokenDetails.fcmToken,
+      fcmPushSet: tokenDetails.fcmPushSet,
+      createTime: Date.now()
+    };
+
+    return this.put(details);
   }
 
   /**
    * This method deletes details of the current FCM token.
    * It's returning a promise in case we need to move to an async
    * method for deleting at a later date.
-   * @return {Promise<Object>} Resolves once the FCM token details have been
-   * deleted and returns the deleted details.
+   *
+   * @return Resolves once the FCM token details have been deleted and returns
+   * the deleted details.
    */
-  deleteToken(token: string): Promise<TokenDetails> {
+  async deleteToken(token: string): Promise<TokenDetails> {
     if (typeof token !== 'string' || token.length === 0) {
       return Promise.reject(
-        this.errorFactory_.create(ERROR_CODES.INVALID_DELETE_TOKEN)
+        this.errorFactory.create(ERROR_CODES.INVALID_DELETE_TOKEN)
       );
     }
 
-    return this.getTokenDetailsFromToken(token).then(details => {
-      if (!details) {
-        throw this.errorFactory_.create(ERROR_CODES.DELETE_TOKEN_NOT_FOUND);
-      }
-
-      return this.openDatabase().then(db => {
-        return new Promise<TokenDetails>((resolve, reject) => {
-          const transaction = db.transaction(
-            [FCM_TOKEN_OBJ_STORE],
-            this.TRANSACTION_READ_WRITE
-          );
-          const objectStore = transaction.objectStore(FCM_TOKEN_OBJ_STORE);
-          const request = objectStore.delete(details.swScope);
-          request.onerror = event => {
-            reject((event.target as IDBRequest).error);
-          };
-          request.onsuccess = event => {
-            if ((event.target as IDBRequest).result === 0) {
-              reject(
-                this.errorFactory_.create(ERROR_CODES.FAILED_TO_DELETE_TOKEN)
-              );
-              return;
-            }
+    const details = await this.getTokenDetailsFromToken(token);
+    if (!details) {
+      throw this.errorFactory.create(ERROR_CODES.DELETE_TOKEN_NOT_FOUND);
+    }
 
-            resolve(details);
-          };
-        });
-      });
-    });
+    await this.delete(details.swScope);
+    return details;
   }
 }
+
+/** Promisifies an IDBRequest. Resolves with the IDBRequest's result. */
+function promisify<ReturnType>(request: IDBRequest): Promise<ReturnType> {
```

Comment:
It's probably worth pulling this function out into a shared utility file so it's not duplicated.

Reply:
I don't think we should do this now. It's not going to be used anywhere else as this is the only file that interacts directly with IndexedDB.